internal/manager/config_validation.go

```diff
@@ -37,7 +37,7 @@ func namespacedNameFromFlagValue(flagValue string) (OptionalNamespacedName, erro
 func gatewayAPIControllerNameFromFlagValue(flagValue string) (string, error) {
 	// https://github.com/kubernetes-sigs/gateway-api/blob/547122f7f55ac0464685552898c560658fb40073/apis/v1beta1/shared_types.go#L448-L463
 	re := regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\/[A-Za-z0-9\/\-._~%!$&'()*+,;=:]+$`)
```

Comment:
to make even further optimization that should give a noticeable performance boost, move the compilation of regular expression outside the function body and check the results

Reply:
Thanks for the suggestion. It has been fixed in 11c06857b6bc7bf47e7fd762c4097d14b22029eb. I have also updated the PR description with the new results. The performance improvement is indeed significant!